test/api/expense_test.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+require "test_helper"
+
+class Fakturoid::Api::ExpenseTest < Fakturoid::TestCase
+  should "get all" do
+    mock_faraday_connection do |stub|
+      response_data = [{ id: 1, number: "N20240201" }]
+      stub.get("expenses.json") { |_env| [200, { content_type: "application/json" }, response_data.to_json] }
+    end
+
+    assert_equal 1, test_client.expense.all.body.size
+  end
+
+  should "search by query" do
+    mock_faraday_connection do |stub|
+      response_data = [{ id: 1, number: "N20240201" }]
+      stub.get("expenses/search.json") { |_env| [200, { content_type: "application/json" }, response_data.to_json] }
```

Comment:
It would be nice if we are able to stub request with params, so we are sure that correct path params are being set. Can we do it?

Reply:
Yes, take a look. Unfortunately you can't pick just query params, you have to test everything like headers and body (if sent). Luckily in the search endpoints we're sending just the query param and some default header set in test helper.